config/VigraConfigureThreading.cmake

```diff
@@ -2,11 +2,20 @@ MACRO(VIGRA_CONFIGURE_THREADING)
     set(THREADING_FOUND 0)
     if(((APPLE AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.7.0") OR MINGW)
          AND NOT WITH_BOOST_THREAD)
-        set(THREADING_FOUND 0)
-        if(${1} MATCHES "REQUIRED")
-            MESSAGE(FATAL_ERROR "Compiler does not support C++ threading. Try to run cmake with '-DWITH_BOOST_THREAD=1'.")
-        else()
-            ADD_DEFINITIONS(-DVIGRA_SINGLE_THREADED)
+        TRY_COMPILE(THREADING_FOUND 
+            ${CMAKE_BINARY_DIR} ${PROJECT_SOURCE_DIR}/config/check_std_thread.cxx
+            OUTPUT_VARIABLE THREADING_TEST_OUTPUT )
+        if (NOT THREADING_FOUND)
+            if(${ARGN} MATCHES "REQUIRED")
+                MESSAGE(FATAL_ERROR "Compiler does not support C++ threading. Try to run cmake with '-DWITH_BOOST_THREAD=1'.\n"
+                "For reference, the failed compiler output is shown below."
+                "${THREADING_TEST_OUTPUT}\n\n"
+                "Compiler does not support C++ threading. Try to run cmake with '-DWITH_BOOST_THREAD=1'.\n"
+                "For reference, the failed compiler output is shown above.\n"
+                )
```

Comment:
I wanted to display the failed compiler output (`THREADING_TEST_OUTPUT`), in case it was useful for the user to figure out *why* his threading implementation was broken, but I was afraid the gobbdelygook from the compiler would obscure the overall diagnostic message.  So, I printed it before and after the output.  But if you think that's confusing, let's just change it.